### PR TITLE
chore: install Node.js in Docker containers (for development mode)

### PR DESCRIPTION
**Pull request by nicomiguelino**
#### Description

* One-up developer experience by installing Node.js in Docker containers, so that they won't have to install Node.js on the host OS.'

> [!NOTE]
> #2060 provides a more elegant solution for conditionally generating Dockerfiles.
> If it gets merged, this PR is still up for more changes.

#### Screenshot(s)

image

### CODE REVIEW

File: docker/Dockerfile.base.tmpl [hunk 1]
```diff
@@ -23,6 +23,8 @@ RUN --mount=type=cache,target=/var/cache/apt \
         lsb-release \
         mplayer \
         net-tools \
+        nodejs \
```
Comment (vpetersson) [inline docker/Dockerfile.base.tmpl:26]: This should be a multi stage build. No reason to keep this around after build.

**Review by vpetersson — CHANGES_REQUESTED** [review]

File: docker/Dockerfile.base.tmpl (hunk 1 above)
Comment (nicomiguelino) [reply docker/Dockerfile.base.tmpl:26]: @vpetersson, in the future, #2060 can help us make changes (even the more sophisticated ones) &mdash; like putting multi-stage builds in templates &mdash; more easily.

File: bin/build_containers.sh [hunk 2]
```diff
@@ -144,6 +142,17 @@ for container in ${SERVICES[@]}; do
...
+        PACKAGES_TO_REMOVE+=(
+            "nodejs"
+            "npm"
+        )
+    fi
+
+    for package in "${PACKAGES_TO_REMOVE[@]}"; do
+        sed "${SED_ARGS[@]}" -e "/$package/d" $(find docker/ -maxdepth 1 -not -name "*.tmpl" -type f)
+    done
+
```
Comment (nicomiguelino) [inline bin/build_containers.sh:155]: @vpetersson, here's my code for removing development dependencies.

**Comment by nicomiguelino** [comment]
@vpetersson, @salmanfarisvp
I'll be closing this pull request since I'll move the changes to #2117.

**Pull request closed without merging**